### DISCUSSION TITLE
List of excluded email domains

### DISCUSSION THREAD

**Discussion by thomasboni** [ory/kratos#1617] [Q&A] [answered]
Hello, 

Is it possible to reject a registration if the email domain is a part of an excluded domains list ? 

I haven't found anything about that in Kratos itself, but I don't know if I can achieve that thanks to `Json Schema` syntax, by specifying a special validation rule on the email field rejecting all emails that we want to ban. 

Thanks

**Comment by aeneasr** [comment]
Currently not, but this could probably be implemented with a web-hook once https://github.com/ory/kratos/pull/1585 is merged

**Reply by thomasboni** [reply]
Ok, thanks for answer !

**Comment by zepatrik** [comment, marked as answer]
I mean you can to that with JSONSchema by specifying something similar to:

```json
{
  "$id": "https://schemas.ory.sh/presets/kratos/quickstart/email-password/identity.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Person",
  "type": "object",
  "properties": {
    "traits": {
      "type": "object",
      "properties": {
        "email": {
          "type": "string",
          "format": "email",
          "title": "E-Mail",
          "minLength": 3,
          "not": {
            "anyOf": [
              {
                "pattern": "@google\\.com$"
              },
              {
                "pattern": "@q-mail\\.com$"
              }
            ]
          },
....
```

but I would consider this more of a workaround than a fix.

**Reply by thomasboni** [reply]
Thanks ! I'm going to use it as a workaround, awaiting #1585
